### DISCUSSION TITLE
Guide on doing Trusted Setup Ceremony for ZK-SNARKs written in Gnark

### DISCUSSION THREAD

**Discussion by shreyas-londhe** [Consensys/gnark#457] [Q&A]
Hi! We at [Nume Crypto](https://www.numecrypto.com) were super happy to use Gnark's high level library to write our zk-snark provers. It enabled us to go from Zero to One. We are now ready for launch and were looking into Trusted Setup ceremonies and how to perform them. Any help on this is much much appreciated. 

Thanks!

cc: @gbotrel

**Comment by gbotrel** [comment]
hi :) gnark doesn't implement a Trusted Setup for Groth16; we may have some external contribution in the near future (we know some individual working on that) towards that direction.

For PlonK + KZG, there are couple of projects / existing ceremonies out there that are compatible. Note that's where we will focus our effort the next few months (improving PlonK support / perf). 

For PlonK + FRI, no trusted setup.

**Reply by shreyas-londhe** [reply]
Ohh I see, Thanks!
